lib/ansible/cli/__init__.py

```diff
@@ -254,6 +254,10 @@ def setup_vault_secrets(loader, vault_ids, vault_password_files=None,
 
         last_exception = found_vault_secret = None
         for vault_id_slug in vault_ids:
+            if not vault_id_slug:
+                # silently ignore empty values
+                continue
+
```

Comment:
I am not sure we want to silently ignore empty IDs, I do agree they should not be used to 'find a file' but that can be short circuited at that bridge.

Reply:
its not always a file name, but a secret, sadly i've found some cases of 'empty secret use', which i should really add tests for if we commit to supporting them ...